frontend/packages/local-storage-operator-plugin/src/features.ts

```diff
@@ -0,0 +1,33 @@
+import { FeatureDetector } from '@console/plugin-sdk';
+import { K8sKind, k8sList } from '@console/internal/module/k8s';
+import { getAnnotations } from '@console/shared/src/selectors/common';
+import { setFlag, handleError } from '@console/internal/actions/features';
+
+export const OCSServiceModel: K8sKind = {
+  label: 'Storage Cluster',
+  labelPlural: 'Storage Clusters',
+  apiVersion: 'v1',
+  apiGroup: 'ocs.openshift.io',
+  plural: 'storageclusters',
+  abbr: 'OCS',
+  namespaced: true,
+  kind: 'StorageCluster',
+  id: 'ocscluster',
+  crd: true,
+};
+const ATTACHED_DEVICES_ANNOTATION = 'cluster.ocs.openshift.io/local-devices';
+export const OCS_ATTACHED_DEVICES_FLAG = 'OCS_ATTACHED_DEVICES';
+
+export const detectOCSAttachedDeviceMode: FeatureDetector = async (dispatch) => {
+  try {
+    const storageClusters = await k8sList(OCSServiceModel, { ns: 'openshift-storage' });
+    const storageCluster = storageClusters.find((sc) => sc.status.phase !== 'Ignored');
+    const isAttachedDevicesCluster =
+      getAnnotations(storageCluster)?.[ATTACHED_DEVICES_ANNOTATION] === 'true';
+    dispatch(setFlag(OCS_ATTACHED_DEVICES_FLAG, isAttachedDevicesCluster));
+  } catch (err) {
+    err?.response?.status === 404
+      ? dispatch(setFlag(OCS_ATTACHED_DEVICES_FLAG, false))
+      : handleError(err, OCS_ATTACHED_DEVICES_FLAG, dispatch, detectOCSAttachedDeviceMode);
+  }
+};
```

Comment:
LSO should not be dependent on OCS. If this is indeed the case, then the plugins that are passed via LSO that use above flag should be passed via the `ceph` plguin. And since ceph is dependent on LSO we can in turn import their flag which are being used for the `AlertActions` below.

Reply:
Sorry, I failed to understand, whats the dependency here ?

Reply:
> LSO should not be dependent on OCS. 
Sorry, I failed to understand, whats the dependency here ?
> If this is indeed the case, then the plugins that are passed via LSO that use above flag should be passed via the `ceph` plguin.
It was like that only and we need to do this change, since it was causing issues, if you check the PR.
> And since ceph is dependent on LSO we can in turn import their flag which are being used for the `AlertActions` below.
Not quite understood.